pkgs/applications/editors/vim/plugins/update.py

```diff
@@ -1,181 +1,177 @@
+    def id(self) -> str:
+        id = self.alias or self.repo.name  # noqa: A001
+        return id.strip().replace(".", "-")
+
+    @t.override
+    async def fetch(self) -> Entry[t.Any]:
+        logger.debug(f"Fetching from GitHub {self.repo.owner}/{self.repo.name}")
+        github_token = os.environ.get("GITHUB_TOKEN")
+
+        if github_token is not None:
+            result = await github_fetch_graphql(
+                self.repo.owner, self.repo.name, github_token
+            )
+        else:
+            result = await github_fetch_rest(
+                self.repo.owner, self.repo.name, github_token=None
+            )
+
+        if self.branch is not None:
+            result = attrs.evolve(result, branch=self.branch, commit=None)
 
-class VimEditor(pluginupdate.Editor):
-    nvim_treesitter_updated = False
+        # TODO: We could also handle redirects like this
+        if (self.repo.owner, self.repo.name) != (result.owner, result.repo):
+            ...
 
-    def generate_nix(
-        self, plugins: List[Tuple[PluginDesc, pluginupdate.Plugin]], outfile: str
-    ):
-        log.info("Generating nix code")
-        log.debug("Loading nvim-treesitter revision from nix...")
-        nvim_treesitter_rev = pluginupdate.run_nix_expr(
-            "(import <localpkgs> { }).vimPlugins.nvim-treesitter.src.rev",
-            self.nixpkgs,
-            timeout=10,
+        result = await result.prefetch_commit(github_token=github_token)
+        prefetched = await nurl.nurl(
+            result.url,
+            revision=result.commit.id if result.commit is not None else None,
+            submodules=result.has_submodules,
         )
+        return MyEntry(info=self, fetched=result, nurl_result=prefetched)
+
+
+@define(frozen=True)
+class MyEntry(Entry[EntryInfo]):
+    info: MyEntryInfo = field(converter=Entry.info_converter(MyEntryInfo))
+    fetched: GHRepository
+    nurl_result: nurl.NurlResult
+
 
-        GET_PLUGINS_LUA = """
-        with import <localpkgs> {};
-        lib.attrNames lua51Packages"""
-        log.debug("Loading list of lua plugins...")
-        luaPlugins = run_nix_expr(GET_PLUGINS_LUA, self.nixpkgs, timeout=30)
-
-        def _isNeovimPlugin(plug: pluginupdate.Plugin) -> bool:
-            """
-            Whether it's a neovim-only plugin
-            We can check if it's available in lua packages
-            """
-            if plug.normalized_name in luaPlugins:
-                log.debug("%s is a neovim plugin", plug)
-                return True
-            return False
-
-        with open(outfile, "w+") as f:
-            log.debug("Writing to %s", outfile)
-            f.write(HEADER)
-            f.write(
-                textwrap.dedent(
-                    """
-                { lib, buildVimPlugin, buildNeovimPlugin, fetchFromGitHub }:
-
-                final: prev:
-                {
-                """
-                )
+@define
+class MyImpl(ABCBase[MyEntry, MyEntryInfo]):
```

Comment:
```suggestion
class Impl(ABCBase[Entry, EntryInfo]):
```
Otherwise the code reads like it got straight copied from the example in the docs.